compaction/task_manager_module.hh

```diff
@@ -366,6 +370,10 @@ public:
         , _exclude_current_version(exclude_current_version)
     {}
 
+    virtual std::string type() const override {
+        return "upgrade " + sstables_compaction_task_impl::type();
```

Comment:
shard upgrade?

Reply:
I do agree that it would be extremely useful to contain a task level in a type, but currently all tasks have the same type on each level (even the repair ones) and I do not think that the change belongs to this PR.

Would you mind if this commit stays as is (differentiating upgrade and scrub compaction) and I will add "shard"/"table" etc. prefix to all tasks as a follow up?

Reply:
Sure, no problem.
Actually, let's demo this and see how it looks and how functionality goes, and then decide if and how to adjust it.